src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -1101,6 +1101,11 @@ private string GetRootJsonContextImplementation()
             {
                 string contextTypeRef = _currentContext.ContextTypeRef;
                 string contextTypeName = _currentContext.ContextType.Name;
+                int backTickIndex = contextTypeName.IndexOf('`');
+                if (backTickIndex != -1)
+                {
+                    contextTypeName = contextTypeName.Substring(0, backTickIndex);
```

Comment:
is there any way to do this without parsing string?

Reply:
I don't see an elegant way.
The `_currentContext.ContextType` is declared as `System.Type`. `System.Type` doesn't have an appropriate API as I know.
Actually `_currentContext.ContextType` is `System.Text.Json.Reflection.TypeWrapper`. `System.Text.Json.Reflection.TypeWrapper` exposes the `SimpleName` property. But I think that cast `_currentContext.ContextType` to `System.Text.Json.Reflection.TypeWrapper` is not good idea and I am not sure that ContextType is always TypeWrapper.
What do you think?